odoo/models.py

```diff
@@ -2608,12 +2607,11 @@ def qualify(field):
         # determine the actual query to execute
         from_clause, where_clause, params = query.get_sql()
         query_str = """ SELECT %(qual_names)s FROM %(from_clause)s
-                        WHERE %(where_clause)s %(order_str)s
```

Comment:
More seriously, I suppose we could avoid changing API semantics by *increasing order* instead of *reducing order* ;-)
@rco-odoo do you think we could make the `BaseModel._fields` an `OrderedDict`, or is the field ordering already determined when the metaclass gets them, before we get a chance to order it?